l2geth/rollup/sync_service.go

```diff
@@ -858,6 +868,38 @@ func (s *SyncService) syncQueueTransactionRange(start, end uint64) error {
 	return nil
 }
 
+// syncTransactions will sync transactions to the remote tip based on the
+// backend
+func (s *SyncService) syncTransactions(backend Backend) (*uint64, error) {
+	getLatest := func() (*uint64, error) {
+		return s.client.GetLatestTransactionIndex(backend)
+	}
+	sync := func(start, end uint64) error {
+		return s.syncTransactionRange(start, end, backend)
+	}
+	index, err := s.sync(getLatest, s.GetNextIndex, sync)
```

Comment:
Stylistic question (or maybe a go thing I'm not aware of) but why use `getLatest` and `sync` here wrapped around the relevant functions? Could we instead pass `backend` through to `s.sync`? Just curious about the design decision. Particularly since you don't do the same thing within `syncQueue`.

Reply:
Its because the third argument to `SyncService.sync` accepts a function with with the following signature:

```golang
func syncer(start, end uint64) error
```

You can see that syncing a transaction range requires the `backend` argument to be passed through. This is done by creating a closure that has the backend argument in scope. The `backend` argument changes the query param that is sent by the `RollupClient` to the DTL. It is either `l1` or `l2` - this allows the client to decided which transaction it wants returned, ie the transaction synced by the DTL from L1 (verifier mode) or from L2 (replica mode).